WatchFaceAlphaKotlin/app/src/main/java/com/example/android/wearable/alpha/AnalogWatchFaceService.kt

```diff
@@ -15,95 +15,82 @@
-import androidx.wear.watchface.style.UserStyleSetting
-import com.example.android.wearable.alpha.data.db.WatchFaceColorStyleEntity
-import com.example.android.wearable.alpha.data.db.analogWatchFaceKeyId
-import com.example.android.wearable.alpha.utils.FRAME_PERIOD_MS_DEFAULT
-import com.example.android.wearable.alpha.utils.createAnalogWatchFace
-import com.example.android.wearable.alpha.utils.createComplicationList
-import com.example.android.wearable.alpha.utils.createUserStyleRepositoryWithCallback
-import com.example.android.wearable.alpha.utils.createUserStyleSettingsList
+import com.example.android.wearable.alpha.data.WatchFaceUserSettingChangesDataSource
+import com.example.android.wearable.alpha.utils.createComplicationSlotManager
+import com.example.android.wearable.alpha.utils.createUserStyleSchema
 import com.example.android.wearable.alpha.viewmodels.AnalogWatchFaceViewModel
-import kotlinx.coroutines.CoroutineScope
-import kotlinx.coroutines.MainScope
-import kotlinx.coroutines.cancel
-import kotlinx.coroutines.launch
 
 /**
  * Handles much of the boilerplate needed to implement a watch face (minus rendering code; see
  * [AnalogWatchCanvasRenderer]) including the complications and settings (styles user can change on
  * the watch face).
  */
 class AnalogWatchFaceService : WatchFaceService() {
-    // Used to launch coroutines (non-blocking way to pull watch face color styles data [only
-    // needed on the initial load of the user color style options]).
-    private val scope: CoroutineScope = MainScope()
-
     private lateinit var analogWatchFaceViewModel: AnalogWatchFaceViewModel
 
-    private var colorStylesList: List<WatchFaceColorStyleEntity>? = null
-
-    override fun onCreate() {
-        super.onCreate()
+    // Used by Watch Face APIs to construct user setting options and repository.
+    override fun createUserStyleSchema(): UserStyleSchema =
+        createUserStyleSchema(context = applicationContext)
 
-        analogWatchFaceViewModel =
-            AnalogWatchFaceViewModel((application as MainApplication).repository)
-
-        // Preloading color styles from database.
-        // TODO: Move to createWatchFace() once it's a suspend function (future alpha).
-        scope.launch {
-            colorStylesList =
-                analogWatchFaceViewModel.getAllWatchFaceColorStyles()
-        }
-    }
+    // Creates all complication user settings and adds them to the existing user settings
+    // repository.
+    override fun createComplicationSlotsManager(
+        currentUserStyleRepository: CurrentUserStyleRepository
+    ): ComplicationSlotsManager = createComplicationSlotManager(
+        context = applicationContext,
+        currentUserStyleRepository = currentUserStyleRepository
+    )
 
     override fun onDestroy() {
+        Log.d(TAG, "onDestroy()")
+
+        // Cancels scope used for retrieving watch face style changes via callbackFlow.
         analogWatchFaceViewModel.clear()
```

Comment:
Since `analogWatchFaceViewModel` is now initialized in `createWatchFace` instead of `onCreate`, is it guaranteed to be initialized in `onDestroy`? We might want to wrap this in `if (::analogWatchFaceViewModel.isInitialized) {`.